chromium/background-scripts/rules.js

```diff
@@ -722,6 +722,36 @@ RuleSets.prototype = {
     return false;
   },
 
+  /**
+   * Get a list of simple rules (active, with no exclusions) for all hosts that
+   * are in a single ruleset, and end in the specified ending.
+   * @param ending Target ending to search for
+   * @returns A list of { host, from_regex, to, scope_regex }
+   */
+  getSimpleRulesEndingWith: function(ending) {
+    let results;
+
+    if (this.wasm_rs) {
+      results = this.wasm_rs.get_simple_rules_ending_with(ending);
+    } else {
+      results = [];
+      for(let [host, rulesets] of this.targets) {
+        if (host.endsWith(ending) &&
+            rulesets.length == 1 &&
+            rulesets[0].active == true &&
+            rulesets[0].exclusions == null
+        ) {
+          for (let rule of rulesets[0].rules) {
+            if (rule.from_c.test("http://" + host + "/")) {
+              results.push({ host, from_regex: rule.from_c.toString(), to: rule.to, scope_regex: rulesets[0].scope.toString() });
```

Comment:
Both `from_regex` and `scope_regex` are String, IMHO it is quite confusing to have the `_regex` part in their names. They are the same as the `from` attribute from a ruleset and `scope` attribute from a ruleset channel, is there any reason not reusing the names?

Reply:
Yeah, so `sendResponse` [must](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage) return a JSON-ifiable object, which is why we convert the RegExp to a string.

Given that this message is being used by an external project, I thought it safer to specify in the key that the `from` and `scope` should be used as a regular expression, thus the `_regex` suffix.